### ISSUE TITLE
GLMakie doesn't plot left-side legend outline when headless

### ISSUE THREAD

**Issue by Octogonapus** [JuliaPlots/Makie.jl#865]
On a headless system, GLMakie doesn't plot the left-side outline for legends. Normally, it does (obviously). Here is an example of what I am talking about:

Normal:
![plot-with-head](https://user-images.githubusercontent.com/4064722/110157612-619b0000-7db6-11eb-8cdc-e9ee8e59088a.png)

Headless:
![plot-headless](https://user-images.githubusercontent.com/4064722/110157611-619b0000-7db6-11eb-86f6-82a194f1f0b0.png)

I am using Julia 1.6.0-rc1, FileIO 1.6, and GLMakie 0.1.28. My OS is 5.10.19-200.fc33.x86_64.

You can find a Docker image that can reproduce the issue in my repository here: https://github.com/Octogonapus/glmakie-legend-bug

**Comment by SimonDanisch** [comment]
Hm, I suspect some subtle differences in the floating point math for the depth buffer... Although these pictures look less clear in that regard then the ones on slack...

**Comment by Octogonapus** [comment]
Would it help if I gathered an rr trace?

**Comment by SimonDanisch** [comment]
Not really... I will need to find some time to see if I can reproduce this and then play around with the depth buffer and see if that's really it

**Comment by Octogonapus** [comment]
Okay. I also used ImageMagick to compare the two images (`compare plot-with-head.png plot-headless.png -compose src diff.png`). This is the diff:

![diff](https://user-images.githubusercontent.com/4064722/110331698-701f2c80-7fed-11eb-95b4-c5fc5ed9efa0.png)

**Comment by awadell1** [comment]
This is happening for me when running on my Mac as well

<img width="411" alt="Screen Shot 2021-07-15 at 2 22 22 PM" src="https://user-images.githubusercontent.com/5857298/125838191-091a9101-2645-476f-9f51-7041d4afa206.png">

Julia: 1.6.2
GLMakie:  v0.4.2
